### ISSUE TITLE
Fail to upgrade to 0.7.x

### ISSUE THREAD

**Issue by avinchakov** [hashicorp/vault#2721]
Hi, I have the issue. My configuration (pseudo HA):

1. two instances, both unsealed, transit mode
2. one remote mysql backend
3. traffic goes to the first vault instance (VIP by keepalived)

I upgraded standby instance - all was ok. But when I upgraded primary instance I got such error:

```
2017/04/26 08:05:23.607726 [INFO ] core: vault is unsealed
2017/04/26 08:05:23.608452 [INFO ] core: post-unseal setup starting
2017/04/26 08:05:23.609326 [INFO ] core: loaded wrapping token key
2017/04/26 08:05:23.611100 [INFO ] core: successfully mounted backend: type=generic path=secret/
2017/04/26 08:05:23.611346 [INFO ] core: successfully mounted backend: type=system path=sys/
2017/04/26 08:05:23.611494 [INFO ] core: successfully mounted backend: type=transit path=transit/
2017/04/26 08:05:23.611566 [INFO ] core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2017/04/26 08:05:23.611620 [ERROR] core: failed to mount entry: path=cubbyhole/ error=cannot mount under existing mount 'cubbyhole/'
2017/04/26 08:05:23.611672 [INFO ] core: pre-seal teardown starting
2017/04/26 08:05:23.611760 [INFO ] core: pre-seal teardown complete
2017/04/26 08:05:23.611796 [ERROR] core: post-unseal setup failed: error=failed to setup mount table
```
I read the branches about a similar problem, but nothing specific for my case was found

**Comment by jefferai** [comment]
You are giving two Vault servers access to the same data store without HA mode. This can and will lead to problems like this.

**Comment by avinchakov** [comment]
This scheme worked perfectly before 0.7x.
It can not be solved?

**Comment by jefferai** [comment]
It seemed to work, or you thought it worked. But it's unsafe. This thread has some more info: https://groups.google.com/d/msgid/vault-tool/dcd1bae6-8722-4da3-92e1-3fc142093a2e%40googlegroups.com

**Comment by avinchakov** [comment]
My configuration really worked and was tested. This is a active/standby. Second instance did not work with the backend until there was traffic on it. 

Is there a way to fix this error in 0.7?

**Comment by jefferai** [comment]
There is no active/standby with MySQL, regardless of whether you are sending traffic to both or not. Vault is stateful and you have two systems performing tasks that only one should be performing. You simply got lucky that things didn't massively break before now, even if they were broken in small ways and you didn't realize it.
